packages/grid/x-data-grid-pro/src/tests/columnPinning.DataGridPro.test.tsx

```diff
@@ -264,12 +264,16 @@ describe('<DataGridPro /> - Column pinning', () => {
   });
 
   describe('dynamic row height', () => {
-    it('should work with dynamic row height', async function test() {
-      if (isJSDOM) {
-        // Need layouting
+    beforeEach(function beforeEach() {
+      const { userAgent } = window.navigator;
+
+      if (isJSDOM || (!userAgent.includes('Headless') && /safari/i.test(userAgent))) {
+        // Need layouting and on Chrome non-headless and Edge these tests are flacky
```

Comment:
Is it `Edge` or `Safari`? 🤔

Reply:
I updated the comment. It's to skip if Edge.